### ISSUE TITLE
[Ubuntu] Clang, GCC and Gfortran versions less than 9.x will be removed from images on March, 29

### ISSUE THREAD

**Issue by MaksimZhukov** [actions/virtual-environments#2950]
**Breaking changes**
Clang, GCC and Gfortran versions less than 9.x will be removed from Ubuntu images

**Target date**
Image deployment will start March, 29 and will take 3-4 days.

**The motivation for the changes**
We are going to deprecate these versions of the listed tools due to lack of free space on Ubuntu images

**Possible impact**
If your project depends on Clang, GCC or Gfortran versions that are less than 9.x — build will be broken.

**Virtual environments affected**
- [x] Ubuntu 16.04
- [x] Ubuntu 18.04
- [x] Ubuntu 20.04
- [ ] macOS 10.13
- [ ] macOS 10.14
- [ ] macOS 10.15
- [ ] macOS 11.0
- [ ] Windows Server 2016 R2
- [ ] Windows Server 2019

**Mitigation ways**
If you need to continue using these versions in your build, please consider the following ways to install it in runtime:

1) **Clang**
```
sudo apt-get update
sudo apt-get install -y "clang-<version>" "lldb-<version>" "lld-<version>" "clang-format-<version>"

# For example, use this command to install clang 8
sudo apt-get install -y "clang-8" "lldb-8" "lld-8" "clang-format-8"
```
2) **GCC**
```
sudo apt-get update
sudo apt-get install g++-<version> -y

# For example, use this command to install gcc 8
sudo apt-get install g++-8 -y
```
3) **Gfortran**
```
sudo apt-get update
sudo apt-get install gfortran-<version> -y

# For example, use this command to install gfortran 8
sudo apt-get install gfortran-8 -y
```

**Comment by vadi2** [comment]
`apt-get install <version> -y` - this doesn't seem valid. Do you mean `apt-get install gcc-<version> -y` ?

**Comment by MaksimZhukov** [comment]
@vadi2, thanks for the note! I have updated the examples

**Comment by miketimofeev** [comment]
The new images were deployed!

**Comment by derekbruening** [comment]
This change seems to have broken 32-bit building.  We install `g++-multilib` and it is suddenly failing, starting ~3 hours ago, with build errors where it finds the wrong libraries, which seems like a toolchain installation problem:

https://github.com/DynamoRIO/dynamorio/pull/4829/checks?check_run_id=2247626216
```
2021-04-01T16:07:58.4961762Z [  2%] Linking C executable bin/linux.infinite
2021-04-01T16:07:58.4962874Z /usr/lib/x86_64-linux-gnu/libdl.so: error adding symbols: File in wrong format
2021-04-01T16:07:58.4964490Z suite/tests/CMakeFiles/linux.infinite.dir/build.make:98: recipe for target 'suite/tests/bin/linux.infinite' failed
2021-04-01T16:07:58.4966284Z CMakeFiles/Makefile2:7501: recipe for target 'suite/tests/CMakeFiles/linux.infinite.dir/all' failed
2021-04-01T16:07:58.4967132Z collect2: error: ld returned 1 exit status

2021-04-01T16:07:58.4971965Z [  2%] Linking C executable bin/client.mbr_instrumentation_segment
2021-04-01T16:07:58.4973167Z /usr/lib/x86_64-linux-gnu/libdl.so: error adding symbols: File in wrong format
2021-04-01T16:07:58.4974711Z suite/tests/CMakeFiles/client.mbr_instrumentation_segment.dir/build.make:98: recipe for target 'suite/tests/bin/client.mbr_instrumentation_segment' failed

2021-04-01T16:09:28.5436729Z [ 58%] Linking CXX executable ../bin32/prefetch_analyzer_launcher
2021-04-01T16:09:28.5438665Z /usr/lib/x86_64-linux-gnu/libz.so: error adding symbols: File in wrong format
2021-04-01T16:09:28.5440235Z clients/drcachesim/CMakeFiles/prefetch_analyzer_launcher.dir/build.make:120: recipe for target 'clients/bin32/prefetch_analyzer_launcher' failed
2021-04-01T16:09:28.5441277Z collect2: error: ld returned 1 exit status
```

**Comment by miketimofeev** [comment]
@derekbruening looks like it's not related to the deprecation. I've tried to change CMake version to 3.19.7 using the following steps and the run passed:
```
    - name: Setup cmake
      uses: jwlawson/actions-setup-cmake@v1.8
      with:
        cmake-version: '3.19.7'
```
Related issue https://github.com/actions/virtual-environments/issues/3078

**Comment by derekbruening** [comment]
> @derekbruening looks like it's not related to the deprecation. I've tried to change CMake version to 3.19.7 using the following steps and the run passed:

Thank you, @miketimofeev!  That indeed seems to fix it, so it does seem to be cmake 3.20.

**Comment by madmongo1** [comment]
Most unwelcome. It cost a day to fix my CI when this little bomb landed.
